spring-cloud-dataflow-server-core/src/test/java/org/springframework/cloud/dataflow/server/batch/AbstractJdbcAggregateJobQueryDaoTests.java

```diff
@@ -0,0 +1,91 @@
+import org.mockito.Mock;
+import org.testcontainers.containers.JdbcDatabaseContainer;
+
+import org.springframework.batch.core.JobInstance;
+import org.springframework.batch.core.JobParameters;
+import org.springframework.batch.core.repository.dao.AbstractJdbcBatchMetadataDao;
+import org.springframework.batch.item.database.support.DataFieldMaxValueIncrementerFactory;
+import org.springframework.cloud.dataflow.core.database.support.DatabaseType;
+import org.springframework.cloud.dataflow.core.database.support.MultiSchemaIncrementerFactory;
+import org.springframework.cloud.dataflow.schema.AppBootSchemaVersion;
+import org.springframework.cloud.dataflow.schema.service.impl.DefaultSchemaService;
+import org.springframework.cloud.dataflow.server.repository.JdbcAggregateJobQueryDao;
+import org.springframework.cloud.dataflow.server.service.JobServiceContainer;
+import org.springframework.jdbc.core.JdbcTemplate;
+
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+abstract class AbstractJdbcAggregateJobQueryDaoTests extends AbstractDaoTests {
+
+	private static final String BASE_JOB_INST_NAME = "JOB_INST_";
+
+	public JdbcSearchableJobInstanceDao jdbcSearchableJobInstanceDao;
+
+	@Mock
+	private JobServiceContainer jobServiceContainer;
+
+	private JdbcAggregateJobQueryDao jdbcAggregateJobQueryDao;
+
+	private DataFieldMaxValueIncrementerFactory incrementerFactory;
+
+	private DatabaseType databaseType;
+
+	protected void prepareForTest(JdbcDatabaseContainer dbContainer,  String schemaName, DatabaseType databaseType) throws Exception {
+		super.prepareForTest(dbContainer, schemaName);
+		this.jdbcAggregateJobQueryDao = new JdbcAggregateJobQueryDao(super.getDataSource(), new DefaultSchemaService(),
+			this.jobServiceContainer);
+		jdbcSearchableJobInstanceDao = new JdbcSearchableJobInstanceDao();
+		jdbcSearchableJobInstanceDao.setJdbcTemplate(new JdbcTemplate(super.getDataSource()));
+		incrementerFactory = new MultiSchemaIncrementerFactory(super.getDataSource());
+		this.databaseType = databaseType;
+	}
+
+	@Test
+	void getJobInstancesForBoot3AndBoot2Instances() throws Exception {
+		assertThat(this.jdbcAggregateJobQueryDao.getJobInstance(1, "boot2")).isNull();
+		assertThat(this.jdbcAggregateJobQueryDao.getJobInstance(1, "boot3")).isNull();
+		createJobInstance("BOOT2", AppBootSchemaVersion.BOOT2);
+		createJobInstance("BOOT3", AppBootSchemaVersion.BOOT3);
+		verifyJobInstance(1, "boot2", "BOOT2");
+		verifyJobInstance(1, "boot3", "BOOT3");
+	}
+
+	private void verifyJobInstance(long id, String schemaTarget, String suffix) throws Exception{
+		JobInstance  jobInstance = this.jdbcAggregateJobQueryDao.getJobInstance(id, schemaTarget);
+		assertThat(jobInstance).isNotNull();
+		assertThat(jobInstance.getJobName()).isEqualTo(BASE_JOB_INST_NAME + suffix );
+	}
+
+	private JobInstance createJobInstance(String suffix, AppBootSchemaVersion appBootSchemaVersion) {
+		String prefix = AbstractJdbcBatchMetadataDao.DEFAULT_TABLE_PREFIX;
```

Comment:
It might be helpful to retrieve a `SchemaVersionTarget` from `org.springframework.cloud.dataflow.schema.service.impl.DefaultSchemaService#getTarget` which then includes the prefixes.